packages/um/package.py

```diff
@@ -329,6 +323,28 @@ def _build_dir(self):
         return join_path(self.stage.source_path, "..", "spack-build")
 
 
+    def patch(self):
+        """Patch the staging directory just before building."""
+        
+        # Get the root to the resources
+        resources_root = join_path(self.stage.source_path, "resources")
+
+        # Optional UM sources (i.e. AM3)
+        if self.spec.variants["um_sources"].value != "NA":
+            self._dynamic_resource(
+                url=self.um_git_url,
+                ref=self.spec.variants["um_sources"].value,
+                dst_dir=join_path(resources_root, "um")
```

Comment:
```suggestion
                dst_dir=env["um_sources"]
```
Could refer to the value set in env to ensure they're identical.